### PR TITLE
ONNX export: Scalar, Reshape - Set appropriate tensor type  …

### PR DESCRIPTION
**Pull request by vandanavk**
## Description ##
np.array calls set default dtype to float64 which is not supported by ONNX. Setting these to float32.

Fixes https://github.com/apache/incubator-mxnet/issues/13044

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Set default dtype to np.float32

## Comments ##
-All tests pass

### CODE REVIEW

**Comment by vandanavk** [comment]
@ankkhedia @anirudhacharya @Roshrini @zhreshold for review

**Comment by ankkhedia** [comment]
@vandanavk Thanks for the contribution! need this to fix models in ONNX Model Zoo.

@mxnet-label-bot [pr-awaiting-review, ONNX]

**Comment by vandanavk** [comment]
Working on @Roshrini's inputs based on offline discussions - in Add, Sub, Mul, Div, tensors can take different types (int32, float32 etc). Therefore, assign type according to input tensor type.

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 1]
```diff
@@ -1840,7 +1840,7 @@ def convert_reshape(node, **kwargs):
     output_shape_list = convert_string_to_list(attrs["shape"])
 
     initializer = kwargs["initializer"]
-    output_shape_np = np.array(output_shape_list)
+    output_shape_np = np.array(output_shape_list, dtype='float32')
```
Comment (zhreshold) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: don't use float for shape

Comment (vandanavk) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: "shape" in reshape seems to be int64 according to ONNX spec. Changing it to int64

**Comment by vandanavk** [comment]
Addressed review comments and rebased. Ready for review again.

**Comment by Roshrini** [comment]
Thanks for fixing this Vandana. LGTM

**Comment by ankkhedia** [comment]
@zhreshold @sandeep-krishnamurthy @nswamy Could you please help with review/merge?

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 1 above)
Comment (luan1412167) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: @zhreshold @vandanavk  output_shape_np = np.array(output_shape_list, dtype='float32'). Is int64 or float32 or float64 right?

**Pull request merged**